pkg/common/resources/resources.go

```diff
@@ -364,6 +364,21 @@ func (r *Resource) SubOnlyExisting(delta *Resource) {
 	}
 }
 
+// AddOnlyExisting adds delta to defined resource.
+// Ignore any type not defined in the base resource (ie receiver).
+func (r *Resource) AddOnlyExisting(delta *Resource) *Resource {
```

Comment:
There is a similar method `SubOnlyExisting`, and it changes the receiver directly. Maybe we should align the behavior. I prefer to return a clone (`AddOnlyExisting`), as we can either reassign it to receiver or keep receiver immutable. WDYT?

Reply:
+1 I think this new method would be the only one which is defined on `*Resource` and also returns something, so it's inconsistent with the existing approaches.

Reply:
I actually like the existing SubOnlyExisting() semantics -- it lets the caller determine whether or not the resource is immuatable (if we're accessing under lock, it's safe to mutate, and saves an expensive Clone()). For values not under lock, Clone() and then modify makes sense.

Reply:
done